WordPress/gutenberg
# Site Editor: Stabilize export endpoint

## Description
Stabilizes site editor export endpoint and updates to use `WP_REST_Controller`.

PR also moves `_remove_theme_attribute_in_block_template_content` into `block-template-utils.php`.

Fixes #36515.

## How has this been tested?
1. Active block theme like TT2
2. Go to Appearance -> Editor (beta)
3. Open the "More tools and Options" dropdown and export.
4. Confirm that the export archive contains the correct files.

All tests are passing.

## Types of changes
Bugfix

## Checklist:
